tests/regression/test_slate_hybridization.py

```diff
@@ -0,0 +1,123 @@
+"""Solve a mixed Helmholtz problem
+
+sigma + grad(u) = 0,
+u + div(sigma) = f,
+u = 0 on boundary,
+
+using hybridisation with SLATE performing the forward elimination.
+The corresponding finite element variational problem:
+
+<tau, sigma> - <div(tau), u> + <<[tau.n], lambda>> = 0 for all tau
+<v, u> + <v, div(sigma)> = <v, f> for all v
+<<gamma, [sigma.n]>> = 0 for all gamma
+
+is solved using broken Raviart-Thomas elements of degree k for
+(sigma, tau), discontinuous Galerkin elements of degree k - 1
+for (u, v), and HDiv-Trace elements of degree k - 1 for (lambda, gamma).
+
+The forcing function is chosen as:
+
+(1+8*pi*pi)*sin(x[0]*pi*2)*sin(x[1]*pi*2),
+
+which reproduces the known analytical solution:
+
+sin(x[0]*pi*2)*sin(x[1]*pi*2)
+"""
+
+
+import pytest
+from firedrake import *
+
+
+@pytest.mark.parametrize("degree", range(1, 4))
+def test_slate_hybridization(degree):
+    # Create a mesh
+    mesh = UnitSquareMesh(8, 8)
+
+    # Create mesh normal
+    n = FacetNormal(mesh)
+
+    # Define relevant function spaces
+    RT = FiniteElement("RT", triangle, degree)
+    BRT = FunctionSpace(mesh, BrokenElement(RT))
+    DG = FunctionSpace(mesh, "DG", degree - 1)
+    T = FunctionSpace(mesh, "HDiv Trace", degree - 1)
+
+    W = BRT * DG
+
+    # Define the trial and test functions
+    sigma, u = TrialFunctions(W)
+    lambdar = TrialFunction(T)
+    tau, v = TestFunctions(W)
+    gammar = TestFunction(T)
+
+    # Define the source function
+    f = Function(DG)
+    f.interpolate(Expression("(1+8*pi*pi)*sin(x[0]*pi*2)*sin(x[1]*pi*2)"))
+
+    # Define finite element variational forms
+    Mass_v = dot(sigma, tau) * dx
+    Mass_p = u * v * dx
+    Div = div(sigma) * v * dx
+    Div_adj = div(tau) * u * dx
+    local_trace = gammar('+') * dot(sigma, n) * dS
+    L = f * v * dx
+
+    # Enforce homogeneous Dirichlet boundary conditions
+    bcs = DirichletBC(T, Constant(0.0), (1, 2, 3, 4))
```

Comment:
The comment is slightly misleading?  I.e. this isn't setting *the solution field* to 0 on the boundary; you're just removing the trace variables from the system since they're not needed/defined on the external boundary of the domain.